layers/+lang/ess/funcs.el

```diff
@@ -9,6 +9,42 @@
 ;;
 ;;; License: GPLv3
 
+
+;; R
+
+(defun spacemacs//ess-r-backend ()
+  "Returns selected backend."
+  ;; backend must be choosed explicitly with this layer
+  (if ess-r-backend
+      ess-r-backend
+    (cond ((configuration-layer/layer-used-p 'lsp) 'lsp)
+          (t 'ess))))
+
+(defun spacemacs//ess-r-setup-backend ()
+  "Conditionally setup ESS backend."
+  (pcase (spacemacs//ess-r-backend)
+    ('lsp (spacemacs//ess-r-setup-lsp))))
+
+(defun spacemacs//ess-r-setup-lsp ()
+  "Setup lsp backend."
+  (if (configuration-layer/layer-used-p 'lsp)
+      (lsp)
+    (message "`lsp' layer is not installed, please add `lsp' layer to your dotfile.")))
+
+(defun spacemacs//ess-r-setup-company ()
+  "Conditionally setup company based on backend."
+  (pcase (spacemacs//ess-r-backend)
+    ('ess (spacemacs//ess-r-setup-company-ess))))
+
+(defun spacemacs//ess-r-setup-company-ess ()
+  "Setup ESS company."
+  (spacemacs|add-company-backends
+    :backends (company-R-library company-R-args company-R-objects :separate)
+    :modes ess-r-mode inferior-ess-r-mode))
+
+
+;; REPL
+
```

Comment:
Please add a REPL declaration to the layer similiar to what has been done in the packages.el in the python layer.
```elisp
      (spacemacs/register-repl 'python
                               'spacemacs/python-start-or-switch-repl "python")
```

Reply:
https://github.com/sei40kr/spacemacs/blob/ffee1c58d560c6037325de4788c2b6ed1931ec61/layers/%2Blang/ess/packages.el#L75

This is currently done here, should this be moved?